### PR TITLE
delete remapping for solmate to reduce remapping.txt requirement for upstream dependents

### PR DESCRIPTION
**Pull request by ewilz**
we'll have some people importing solmate/permit2 indirectly from node based projects. This is an experiment with configuring things as to not force them to have a remappings.txt file for their submodules. feel free to close though

### CODE REVIEW

**Review by marktoda — CHANGES_REQUESTED** [review]
Undo submodule updates, otherwise seems fine to me

File: remappings.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
 forge-std/=lib/forge-std/src/
 openzeppelin-contracts/=lib/openzeppelin-contracts/
-solmate/=lib/solmate/src/
+solmate/=lib/solmate/
```
[the remark this answers is not in the document]
Comment (ewilz) [reply remappings.txt]: had some trouble with this

File: .gas-snapshot [hunk 2]
```diff
@@ -102,4 +102,4 @@ TypehashGeneration:testPermitTransferFrom() (gas: 36520)
 TypehashGeneration:testPermitTransferFromWithWitness() (gas: 43369)
 TypehashGeneration:testPermitTransferFromWithWitnessIncorrectPermitData() (gas: 43430)
 TypehashGeneration:testPermitTransferFromWithWitnessIncorrectTypehashStub() (gas: 43833)
-MockPermit2Lib:testPermit2Code(address):(bool) (runs: 256, μ: 2911, ~: 2911)
+MockPermit2Lib:testPermit2Code(address):(bool) (runs: 256, μ: 35465847065545473, ~: 2911)
```
Comment (marktoda) [inline .gas-snapshot:105]: waaaaat lol

**Review by marktoda — APPROVED** [review]

File: .gas-snapshot (hunk 2 above)
Comment (ewilz) [reply .gas-snapshot:105]: does this need to be investigated?

**Comment by siobh9** [comment]
Am doing some work on [Juicebox Contracts v4](https://github.com/Bananapus/juice-contracts-v4/pull/54), which have this repo as a dependency, to get them fully using forge, but am running into an error where `PermitSignature.sol`'s `openzeppelin-contracts/contracts/utils/cryptography/draft-EIP712.sol` import is attempting to read that file from the top level `lib/openzeppelin-contracts` folder instead of the `lib/permit2/lib/openzeppelin-contracts` folder.

The simplest answer to resolve this seems to be that the `remapping.txt` file in this repo should explicitly specify where this import should look, and so wanted to bring this up on the PR where y'all got rid of this, but I was curious if y'all have seen this issue at all and/or know of any alternatives I should try here to get the mapping to work correctly.

Thank you!

image

UPDATE: figured out a hack for this [here](https://github.com/Bananapus/juice-contracts-v4/pull/54/commits/5880ad9bde67ecb178fedf5d55abdb81e4508f59) but definitely not optimal/is not a scalable solution if other dependencies are also not fully remapped.

File: remappings.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Tomiborges) [reply remappings.txt]: Apaga tudo

[the remark this answers is not in the document]
Comment (Tomiborges) [reply remappings.txt]: Apaga tudo

**Pull request merged**